src/basic/calendarspec.c

```diff
@@ -197,6 +197,10 @@ _pure_ static bool chain_valid(CalendarComponent *c, int from, int to, bool end_
         if (c->start < from || c->start > to)
                 return false;
 
+        /* Avoid overly large values that could cause overflow */
+        if (c->repeat > to - from)
+                return false;
```

Comment:
hmm, if we calc "to - from" now, we should maybe also add an explicit assert before that assures that "to >= from" is. I mean, it is in all calls we make to this function, but we should denote that in code I am sure, just to be safe.